torch/_dynamo/utils.py

```diff
@@ -1043,6 +1048,9 @@ def same(
             # Check error from fp64 version
             if fp64_ref.dtype == torch.float64:
                 ref_error = rmse(fp64_ref, ref).item()
+                if math.isnan(ref_error):
+                    return True
```

Comment:
Does this skip checking the entire tensor if there is a nan anywhere in the ref? So say I had:
```
ref = tensor([1, nan, 2, 3, 4])
res = tensor([5, 4, 3, 2, 1])
```
We would record this as a pass?

Reply:
Right. I think it's pretty unlikely that one output is nan and the others are well formed. It's much more likely that we were running in too low of precision to correctly compute this particular output.

I updated the conditions slightly, so that we'll only return True if fp64 result had no nans.

The goal here is for us to not chase down spurious numerics differences. Id also be happy throwing on nan result. What do you suggest ?

Reply:
Throwing on an unexpected nan makes sense. When the `equal_nan` argument is set, it would also make sense to create a mask where the values are nan in the ref and mask those out before computeing the `rmse`.

Reply:
Changed to warning. We could make further changes.